Userland/Libraries/LibGUI/Dialog.h

```diff
@@ -33,11 +33,12 @@ class Dialog : public Window {
     virtual void close() override;
 
 protected:
-    explicit Dialog(Window* parent_window);
+    explicit Dialog(Window* parent_window, bool center_within_parent_window = true);
```

Comment:
I like the idea of being able to control the positioning, especially in this case, but perhaps we could use an enum instead for clarity? Passing false does not really tell us it will be centered on screen even though it's the default behaviour. Instead we could use something like Dialog::ScreenPosition::CenterWithinParent / CenterOnScreen. It could also be extended to ::TopLeft, ::Right etc in the future. Just a thought!

Reply:
That sounds cool, I just force pushed with new implementation that does everything you mentioned.